aria_data_loaders/aria_data_utils/aria_live_streamer.py

```diff
@@ -0,0 +1,469 @@
+        Returns the average transformation of aria world frame to marker frame
+
+        We get a device_T_marker for each frame in which marker is detected.
+        Depending on the frame rate of image capture, multiple frames may have captured the marker.
+        Averaging all transforms would be best way to compensate for any noise that may exist in any frame's detections
+        camera_T_marker is used to compute device_T_marker[i] and thus ariaWorld_T_marker[i].
+        Then we average all ariaWorld_T-marker to find average marker pose wrt ariaWorld.
+
+        NOTE: To compute average of SE3 matrix, we find the average of translation and rotation separately.
+              The average rotation is obtained by averaging the quaternions.
+        NOTE: Since multiple quaternions can represent the same rotation, we ensure that the 'w' component of the
+              quaternion is always positive for effective averaging.
+
+        Args:
+            img_metadata_list (List): List of image metadata
+            device_T_marker_list (List): List of Sophus SE3 transforms from Device frame to marker
+            filter_dist (float, optional): Distance threshold for valid detections. Defaults to FILTER_DIST.
+        """
+        ariaWorld_T_marker = ariaWorld_T_device * device_T_marker
+        marker_position = ariaWorld_T_marker.translation()
+        device_position = ariaWorld_T_device.translation()
+        delta = marker_position - device_position
+        dist = np.linalg.norm(delta)
+
+        # Consider only those detections where detected marker is within a certain distance of the camera
+        if dist < filter_dist:
+            # If the number of detections exceeds the fix list length, remove the first element
+            if len(self.marker_positions_list) >= self.fix_list_length:
+                self.marker_positions_list.pop(0)
+                self.marker_quaternion_list.pop(0)
+
+            self.marker_positions_list.append(marker_position)
+            quat = Rotation.from_matrix(ariaWorld_T_marker.rotationMatrix()).as_quat()
+
+            # Ensure quaternion's w is always positive for effective averaging as multiple quaternions can represent the same rotation
+            if quat[3] > 0:
+                quat = -1.0 * quat
+            self.marker_quaternion_list.append(quat)
+
+            # Compute the average transformation as new data got appended
+            self.avg_ariaWorld_T_marker = compute_avg_spSE3_from_nplist(
+                a_T_b_position_list=self.marker_positions_list,
+                a_T_b_quaternion_list=self.marker_quaternion_list,
+            )
+
+        return self.avg_ariaWorld_T_marker
+
+
+@click.command()
+@click.option("--do-update-iptables", is_flag=True, type=bool, default=False)
+@click.option("--read-only", is_flag=True, type=bool, default=False)
+@click.option("--debug", is_flag=True, type=bool, default=False)
+def main(do_update_iptables: bool, read_only: bool, debug: bool):
```

Comment:
quick docstring to explain what main does.

Reply:
added